app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -7,7 +7,7 @@ module TwoFactorAuthenticatable
     before_action :require_current_password, if: :current_password_required?
     before_action :check_already_authenticated
     before_action :reset_attempt_count_if_user_no_longer_locked_out, only: :create
-    before_action :apply_secure_headers_override, only: :show
+    before_action :apply_secure_headers_override, only: [:show, :create]
```

Comment:
I think the CSP headers need to be rendered at the last page with user interaction before a redirect (ex if we render a page with a form, submit the form, and then redirect to 2 or 3 pages, the CSP headers to allow any of those redirects outside `self` need to be on the page with the form).

The difference between OIDC and SAML is that for SAML's last "hop" we render a page with a form and auto-submit that form via JS, which I think gets around the CSP issue since the page with the form is rendered.

There is no such form for OIDC (and I don't think it makes sense to rely on JS for a redirect) so I expanded the approach we used to allow skipping the "allow/deny" page entirely in the first place.